.github/workflows/build_deploy_exm_fe.yml

```diff
@@ -0,0 +1,64 @@
+name: Build and Deploy EXM FE
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - name: Install dependencies
+      run: npm install
+      working-directory: expenses-manager-frontend/src/main/frontend
+
+    - name: Build Angular application
+      run: npm run build --prod
+      working-directory: expenses-manager-frontend/src/main/frontend
+
+    - name: Azure CLI Login
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Get VM Public IP
+      id: get-vm-ip
+      run: |
+        az vm list-ip-addresses --name ${{ secrets.EXM_VM_NAME }} --resource-group ${{ secrets.RESOURCE_GROUP_NAME }} --query "[0].virtualMachine.network.publicIpAddresses[0].ipAddress" -o tsv
+      continue-on-error: true
+
+    - name: SSH into VM and install Nginx
+      run: |
+        ssh -o StrictHostKeyChecking=no $admin_username@${{ steps.get-vm-ip.outputs.vm_ip }} << 'EOF'
```

Comment:
asta nu o sa iti mearga pentru ca nu folosesti cheia pentru autentificare si autentificarea cu parola e oprita pe servere (cred si sper) (vezi `-i` [aici](https://man.openbsd.org/ssh#i)). pe langa asta din cate stiu eu nu ai metoda sigura prin care sa pui cheia privata direct in comanda, va trebui facuta o variabila pentru cheie si sa pui continutul ei intr-un fisier temporar pe care sa il folosesti in comanda.